tools/swift-inspect/Sources/SwiftInspectHeapWalk/SwiftInspectHeapWalk.cpp

```diff
@@ -0,0 +1,184 @@
+#if defined(_WIN32)
+
+#pragma comment(lib, "swiftCore.lib")
+
+#include <windows.h>
+#include <stdio.h>
+
+namespace {
+struct HeapEntry {
+  uintptr_t address;
+  uintptr_t size;
+};
+} // anonymous namespace
+
+#define BUF_SIZE 512
+#define BUF_NUM_ENTRIES (BUF_SIZE / sizeof(HeapEntry))
+#define SHARED_MEM_NAME "Local\\SwiftInspectFileMapping"
+#define READ_EVENT_NAME "Local\\SwiftInspectReadEvent"
+#define WRITE_EVENT_NAME "Local\\SwiftInspectWriteEvent"
+#define WAIT_TIMEOUT_MS 30000
+
+static int heapWalk() {
+  HANDLE hMapFile = OpenFileMappingA(FILE_MAP_ALL_ACCESS, false,
+                                     SHARED_MEM_NAME);
+  if (hMapFile == NULL) {
+    printf("OpenFileMapping failed\n");
+    return 1;
+  }
+  HeapEntry *buf = (HeapEntry *)
+    MapViewOfFile(hMapFile,
+                  FILE_MAP_ALL_ACCESS,
+                  0,
+                  0,
+                  BUF_SIZE);
+  if (buf == NULL) {
+    printf("MapViewOfFile failed\n");
+    CloseHandle(hMapFile);
```

Comment:
Since this is C++ code, would you mind adding in a `scoped_handle`?  (See https://github.com/apple/swift-installer-scripts/commit/927c674fb3278d2e61a63e3d7c8e7efd0ff3aa2e#diff-baf3457e394df6903a81ffffaaf57a045306145a255ce7c5a5b34ba01e7f7d56L25-L33 if you would like a reference)